python/cuml/test/test_metrics.py

```diff
@@ -160,6 +162,121 @@ def test_rand_index_score(name, nrows):
+        a = np.array(ground_truth, dtype=np.int32)
+        b = np.array(predictions, dtype=np.int32)
+
+        a_dev = cuda.to_device(a)
+        b_dev = cuda.to_device(b)
+
+        handle, stream = get_handle(use_handle)
+
+        return cuml.metrics.homogeneity_score(a_dev, b_dev, handle=handle)
+
+    # Perfect labelings are homogeneous
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [1, 1, 0, 0]),
+                                   1.0, decimal=4)
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 0, 1, 1]),
+                                   1.0, decimal=4)
+
+    # Non-perfect labelings that further split classes into more clusters can
+    # be perfectly homogeneous
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 0, 1, 2]),
+                                   1.0, decimal=4)
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 1, 2, 3]),
+                                   1.0, decimal=4)
+
+    # Clusters that include samples from different classes do not make for an
+    # homogeneous labeling
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 1, 0, 1]),
+                                   0.0, decimal=4)
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [0, 0, 0, 0]),
+                                   0.0, decimal=4)
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_homogeneity_score_big_array(use_handle):
+    def assert_equal_sklearn(random_generation_lambda):
+        rng = np.random.RandomState(1234)  # makes it reproducible
+        a = random_generation_lambda(rng)
+        b = random_generation_lambda(rng)
+
+        a_dev = cuda.to_device(a)
+        b_dev = cuda.to_device(b)
+
+        handle, stream = get_handle(use_handle)
+
+        score = cuml.metrics.homogeneity_score(a_dev, b_dev, handle=handle)
+        ref = sk_hom_score(a_dev, b_dev)
+
+        np.testing.assert_almost_equal(score, ref, decimal=4)
+
+    assert_equal_sklearn(lambda rng: rng.randint(0, 1000, int(10e4),
+                                                 dtype=np.int32))
+    assert_equal_sklearn(lambda rng: rng.randint(-1000, 1000, int(10e4),
+                                                 dtype=np.int32))
+
+
+def assert_mi_equal_sklearn(a, b, use_handle):
+    a_dev = cuda.to_device(a)
+    b_dev = cuda.to_device(b)
+
+    handle, stream = get_handle(use_handle)
+
+    score = cuml.metrics.mutual_info_score(a_dev, b_dev, handle=handle)
+    ref = sk_mi_score(a_dev, b_dev)
+    np.testing.assert_almost_equal(score, ref, decimal=4)
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_mutual_info_score(use_handle):
+    def assert_ours_equal_sklearn(ground_truth, predictions):
+        a = np.array(ground_truth, dtype=np.int32)
+        b = np.array(predictions, dtype=np.int32)
+        assert_mi_equal_sklearn(a, b, use_handle=use_handle)
+
+    assert_ours_equal_sklearn([0, 0, 1, 1], [1, 1, 0, 0])
```

Comment:
I think this list would be better as a parameterization (pytest.mark.parameterize("input_data", [ ... ])

 rather than all asserts in a single tests, since then you get one failure that clearly shows which test variant was failing and you can choose to e.g. rerun just that test.

Reply:
Agreed.